src/statsmodels/formula.jl

```diff
@@ -62,7 +62,7 @@ type ModelFrame
     contrasts::Dict{Symbol, ContrastsMatrix}
 end
 
-@compat const AbstractFloatMatrix{T<:AbstractFloat} = AbstractMatrix{T}
+@compat const AbstractFloatMatrix{T<:Union{AbstractFloat,NAtype}} = AbstractMatrix{T}
```

Comment:
The name of the alias is not completely correct with this change. Maybe you could just remove that variable and use `ModelMatrix{S<:Union{AbstractFloat,NAtype}, T<:AbstractMatrix{S}}` below now that it's supported?